lib/netsuite/actions/abstract_action.rb

```diff
@@ -0,0 +1,32 @@
+module NetSuite
+  module Actions
+    class AbstractAction
+      def request(credentials={})
+        NetSuite::Configuration.connection(request_options, credentials, soap_header_extra_info).call(action_name, message: request_body)
+      end
+
+      protected
+
+      def action_name
+        raise NotImplementedError, 'Not implemented on abstract class'
+      end
+
+      def initialize
+        raise NotImplementedError, 'Not implemented on abstract class'
+      end
+
+      def request_body
+        raise NotImplementedError, 'Not implemented on abstract class'
+      end
+
+      def request_options
```

Comment:
is there a reason we wouldn't default to `{}` and omit these dup methods on the specific actions?